tests/tensorflow2/test_keras_gradtape.py

```diff
@@ -0,0 +1,252 @@
+# Third Party
```

Comment:
Add docstrings to all tests describing expected behavior and what is being tested. It's tough to follow what's going on with all the test parameters and conditional paths.

Reply:
done in the more recent commits. 

also, had to move up gradtape test to run before fit() tests. there's something going on in the fit() execution that makes the gradtape tests fail if run after fit(). I've tried `tf.reset_default_graph()`, `tf.keras.backend.clear_session()` (FWIW), `tf.enable_eager_execution()` but none worked.